tests/integration/test_ingress.py

```diff
@@ -12,23 +12,77 @@
 from juju.model import Model
 
 import jenkins
+import state
+from charm import AGENT_DISCOVERY_INGRESS_RELATION_NAME
 
 
 @pytest.mark.abort_on_fail
 async def test_ingress_integration(
-    model: Model, application: Application, ingress_related: Application, external_hostname: str
+    model: Model,
+    application: Application,
+    traefik_application: Application,
+    external_hostname: str,
 ):
     """
     arrange: deploy the Jenkins charm and establish relations via ingress.
     act: send a request to the ingress in /.
     assert: the response succeeds.
     """
-    status = await model.get_status(filters=[application.name])
-    unit = next(iter(status.applications[application.name].units))
-    address = status["applications"][application.name]["units"][unit]["address"]
+    await application.relate("ingress", traefik_application.name)
+    await model.wait_for_idle(
+        apps=[application.name, traefik_application.name], wait_for_active=True, timeout=20 * 60
+    )
+    # Find traefik IP
+    status = await model.get_status(filters=[traefik_application.name])
+    unit = next(iter(status.applications[traefik_application.name].units))
+    traefik_address = status["applications"][traefik_application.name]["units"][unit]["address"]
```

Comment:
This seems to be duplicated below and could be extracted into a function.

Reply:
makes sense, I'll see if I can put it in the traefik charm fixture

Reply:
added it in the traefik charm's fixture